contract/src/test/java/io/neow3j/contract/NFTokenTest.java

```diff
@@ -0,0 +1,204 @@
+        account2 = new Account(ECKeyPair.create(
+                Numeric.hexStringToByteArray(
+                        "b4b2b579cac270125259f08a5f414e9235817e7637b9a66cfeb3b77d90c8e7f9")));
+        account3 = new Account(ECKeyPair.create(
+                Numeric.hexStringToByteArray(
+                        "3a100280baf46ea7db17bc01b53365891876b4a2db11028dbc1ccb8c782725f8")));
+    }
+
+    @Test
+    public void testTransfer() throws IOException {
+        setUpWireMockForInvokeFunction("decimals", "nft_decimals_0.json");
+        setUpWireMockForInvokeFunction("ownerOf", "nft_ownerof.json");
+        setUpWireMockForInvokeFunction("transfer", "nft_transfer.json");
+
+        byte[] expectedScript = new ScriptBuilder().contractCall(NF_TOKEN_SCRIPT_HASH, TRANSFER,
+                Arrays.asList(
+                        ContractParameter.hash160(account1.getScriptHash()),
+                        ContractParameter.hash160(TOKEN_ID)))
+                .toArray();
+
+        Wallet wallet = Wallet.withAccounts(account1);
+        TransactionBuilder b = nfTestToken.transfer(wallet, account1.getScriptHash(), TOKEN_ID);
+        assertThat(b.getScript(), is(expectedScript));
+    }
+
+    @Test
+    public void testTransfer_Divisible() throws IOException {
+        setUpWireMockForInvokeFunction("decimals", "nft_decimals_5.json");
+
+        exceptionRule.expect(IllegalStateException.class);
+        exceptionRule.expectMessage("method is only implemented on NF tokens that are divisible.");
+        nfTestToken.transfer(Wallet.create(), account1.getScriptHash(), TOKEN_ID);
+    }
+
+    @Test
+    public void testTransfer_MultipleOwners() throws IOException {
+        setUpWireMockForInvokeFunction("decimals", "nft_decimals_0.json");
+        setUpWireMockForInvokeFunction("ownerOf", "nft_ownerof_multiple.json");
+
+        exceptionRule.expect(IllegalStateException.class);
+        exceptionRule.expectMessage("has 2 owners. To transfer fractions use the method " +
+                "transferFractions.");
+        nfTestToken.transfer(Wallet.create(), account1.getScriptHash(), TOKEN_ID);
+    }
+
+    @Test
+    public void testTransfer_WalletDoesNotContainTokenOwner() throws IOException {
+        setUpWireMockForInvokeFunction("decimals", "nft_decimals_0.json");
+        setUpWireMockForInvokeFunction("ownerOf", "nft_ownerof.json");
+
+        exceptionRule.expect(IllegalArgumentException.class);
+        exceptionRule.expectMessage("The provided wallet does not contain the account");
+        nfTestToken.transfer(Wallet.withAccounts(account2), account1.getScriptHash(), TOKEN_ID);
+    }
+
+    @Test
+    public void testTransferFraction() throws IOException {
+        setUpWireMockForInvokeFunction("decimals", "nft_decimals_5.json");
+        setUpWireMockForInvokeFunction("ownerOf", "nft_ownerof.json");
+        setUpWireMockForInvokeFunction("transfer",
```

Comment:
This mock is not necessary in this test. Removing this line also makes the nft_transferfractions.json obsolete.